include/glow/Runtime/HostManager/HostManager.h

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef GLOW_RUNTIME_HOST_MANAGER_H
+#define GLOW_RUNTIME_HOST_MANAGER_H
+#include "glow/Graph/Graph.h"
+
+#include <llvm/ADT/StringRef.h>
+
+#include <unordered_map>
+namespace glow {
+class DependencyGraph;
+class DeviceManager;
+class Partitioner;
+class Provisioner;
+class Executor {}; // temporary the Executor header will need to be included
+class Context;
+class HostManager final {
+  int activeCount_;
+  int totalCount_;
+  std::unordered_map<int, DependencyGraph> networks_;
```

Comment:
@yinghai networks can be added and removed but their Ids (the keys in this map) are monotonically increasing so a vector could end up being sparse and unnecessarily wasting space

Reply:
We remove network because high level application wants to retire certain model. From c2 level, `onnxReleaseGraph` is called and corresponding graph hit refcouter==0.

Reply:
yes, agree